data/creaturescripts/scripts/playerdeath.lua

```diff
@@ -81,7 +81,8 @@ function onDeath(player, corpse, killer, mostDamageKiller, lastHitUnjustified, m
 				end
 
 				if warId ~= false then
-					db.asyncQuery("INSERT INTO `guildwar_kills` (`killer`, `target`, `killerguild`, `targetguild`, `time`, `warid`) VALUES (" .. db.escapeString(killerName) .. ", " .. db.escapeString(player:getName()) .. ", " .. killerGuild .. ", " .. targetGuild .. ", " .. os.time() .. ", " .. warId .. ")")
+					local playerName = player:getName()
```

Comment:
All is fine, but why should you define a variable with the player name to use in formatting the string passed to the `db.asyncQuery` function? I think it does not affect anything to leave it as it was, if there is a reason for this, I really want to know.

Reply:
@MillhioreBT db.asyncQuery is not executed instantly, so the player might not "exist" anymore, the same issue from addEvent I guess

Reply:
There is no doubt that the function does not execute immediately, but the parameter yes it does, you can enter an print("hello") and check that the string is formatted immediately when the function is called.

Reply:
you are not passing player userdata, but string, string won't get nulled anywhere. Using player:getName() is perfectly fine here.